src/main/java/teammates/storage/sql/EntitiesDb.java

```diff
@@ -70,17 +67,8 @@ private A createEntity(A entityToAdd, boolean shouldCheckExistence)
 
         E entity = convertToEntityForSaving(entityToAdd);
 
-        Transaction tx = null;
-
-        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            tx = session.beginTransaction();
-            session.persist(entity);
-            tx.commit();
-            log.info("Entity created: " + JsonUtils.toJson(entityToAdd));
-        } catch (HibernateException e) {
-            // TODO: Handle errors
-            tx.rollback();
-        }
+        HibernateUtil.getSessionFactory().getCurrentSession().persist(entity);
+        log.info("Entity created: " + JsonUtils.toJson(entityToAdd));
```

Comment:
With the new servlet filter, the logs here are no longer accurate as the messages should only be logged after the transaction has been committed.